tests/src/test/scala/it/unibo/scafi/test/functional/ScafiTestUtils.scala

```diff
@@ -96,24 +96,32 @@ object ScafiTestUtils {
 
   def manhattanNet(
                     side: Int = 3,
-                    northWestDetached: Boolean = false,
-                    northEastDetached: Boolean = false,
-                    southWestDetached: Boolean = false,
-                    southEastDetached: Boolean = false
+                    step: Int = 1,
+                    rng: Double = 1.5,
+                    detachedNodesCords: Set[(Int, Int)] = Set()
                   ): Network with SimulatorOps = {
-    /*simulatorFactory.gridLike(GridSettings(side, side, 1, 1,
-    mapPos = (a,b,px,py) => if(southWestDetached && a==side-1 && b==side-1) (Int.MaxValue, Int.MaxValue) else (px,py)), rng = 1.5)
-     */
-    val sideAdj = side - 1
+    var lastDetachedPosition: (Double, Double) = (Int.MaxValue, Int.MaxValue)
     simulatorFactory.gridLike(
       GridSettings(
-        side, side, 1, 1,
-        mapPos = (a,b,px,py) => (a,b,px,py) match {
+        side, side, step, step,
+        mapPos = (a,b,px,py) => {
+          if (detachedNodesCords contains (a,b)) {
```

Comment:
Maybe you can write this more concisely via something as `detachedNodesCoords.get(a,b).map(...).getOrElse((px,py))

Reply:
In order to use `get(a,b)` i have to convert `detachedNodesCords` to a map

```scala
 detachedNodesCoords
            .zipWithIndex
            .toMap
            .get(a,b)
            .map{_ =>
              lastDetachedPosition = (lastDetachedPosition._1 - rng, lastDetachedPosition._2 - rng)
              lastDetachedPosition
            }
            .getOrElse[(Double, Double)](px, py)
```

What do you think?

Reply:
What about `find`? Just use something to turn the if/else into a single more concise expression